test_linked_list.py

```diff
@@ -0,0 +1,106 @@
+# _*_ encoding: utf-8 _*_
+"""Test linked_list.py."""
+
+
+def test_new_list():
+    """Test list constructor."""
+    from linked_list import LinkedList
+    new_list = LinkedList()
+    assert isinstance(new_list, LinkedList)
+
+
+def test_insert():
+    """Test insert method."""
+    from linked_list import LinkedList
+    new_list = LinkedList()
+    test_size = new_list.size()
+    new_list.insert("test")
+    assert test_size < new_list.size()
+
+
+def test_insert_list():
+    """Test insert method with a list."""
+    from linked_list import LinkedList
+    new_list = LinkedList()
+    new_list.insert([1, 2, 3])
+    assert new_list.size() == 3
+
+
+def test_size():
+    """Test size method."""
+    from linked_list import LinkedList
+    new_list = LinkedList()
+    new_list.insert("test")
+    new_list.insert("kickass test")
+    new_list.insert("negasonic ultra test")
+    size = new_list.size()
+    assert size == 3
+
+
+def test_pop():
+    """Test pop method."""
+    from linked_list import LinkedList
+    first_list = LinkedList()
+    first_list.insert([1, 2, 3, 4])
+    old_size = first_list.size()
+    popped = first_list.pop()
+    new_size = first_list.size()
+    assert new_size < old_size
+    assert popped.get_data() == 1
+
+
+def test_search():
+    """Test search method."""
+    from linked_list import LinkedList
+    new_list = LinkedList()
+    new_list.insert([1, 2, 3, 4])
+    query = new_list.search(3)
+    assert query.get_data() == 3
+
+
+def test_remove():
+    """Test remove method."""
+    from linked_list import LinkedList
+    new_list = LinkedList()
+    new_list.insert([1, 2, 3, 4])
+    new_list.remove(new_list.search(3))
+    assert new_list.size() == 3
```

Comment:
Just a random thought... But instead of testing the size of after removal, perhaps something like this would be a better test 
```  assert new_list.display() == '(1, 2, 4)' ```
That way you are testing that remove indeed removed the node you were expecting it to remove.